src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -230,6 +230,9 @@ export class TerminalService implements ITerminalService {
 			env[`PTYSHELLARG${i}`] = arg;
 		});
 		env['PTYCWD'] = this.contextService.getWorkspace() ? this.contextService.getWorkspace().resource.fsPath : os.homedir();
+		if (!env['LANG'] && platform.locale) {
+			env['LANG'] = this.getLang(platform.locale);
```

Comment:
'locale' will show the LC_ variables set the same, except for LC_ALL which will be empty. That's the same as you're screenshot shows and the same as the terminal shows me under OS X. I assume that should be fine.